deps/rabbitmq_aws/src/rabbitmq_aws.erl

```diff
@@ -546,19 +546,27 @@ api_get_request(Service, Path) ->
   api_get_request_with_retries(Service, Path, ?MAX_RETRIES, ?LINEAR_BACK_OFF_MILLIS).
 
 
-  -spec api_get_request_with_retries(string(), path(), integer(), integer()) -> result().
-  %% @doc Invoke an API call to an AWS service with retries.
-  %% @end
+-spec api_get_request_with_retries(string(), path(), integer(), integer()) -> result().
+%% @doc Invoke an API call to an AWS service with retries.
+%% @end
 api_get_request_with_retries(_, _, 0, _) ->
   rabbit_log:warning("Request to AWS service has failed after ~b retries", [?MAX_RETRIES]),
   {error, "AWS service is unavailable"};
 api_get_request_with_retries(Service, Path, Retries, WaitTimeBetweenRetries) ->
   ensure_credentials_valid(),
   case get(Service, Path) of
-    {ok, {_Headers, Payload}} -> rabbit_log:debug("AWS request: ~s~nResponse: ~p", [Path, Payload]),
-                                 {ok, Payload};
-    {error, {credentials, _}} -> {error, credentials};
-    {error, Message, _}       -> rabbit_log:warning("Error occurred ~s~nWill retry AWS request, remaining retries: ~b", [Message, Retries]),
-                                 timer:sleep(WaitTimeBetweenRetries),
-                                 api_get_request_with_retries(Service, Path, Retries - 1, WaitTimeBetweenRetries)
+    {ok, {_Headers, Payload}} ->
+          rabbit_log:debug("AWS request: ~s~nResponse: ~p", [Path, Payload]),
+          {ok, Payload};
+    {error, {credentials, _}} ->
+          {error, credentials};
+    {error, Message, Response} ->
+          rabbit_log:warning("Error occurred ~s~nWill retry AWS request, remaining retries: ~b", [Message, Retries]),
+          case Response of
+              {_Headers, Payload} -> rabbit_log:warning("Response: ~p", [Payload]);
```

Comment:
We discussed logging the request-id to help with debugging. But it appears that different AWS services return the request ID differently. For example, S3 returns "x-amz-request-id" header, whereas APIG returns "x-amzn-RequestId".

Logging the response body will provide sufficient information for debugging, but it might be noisy at "warning" level. However, logging at "debug" level will prevent us from debugging common transient issues.

Alternatively, we could try to log any headers with "request-id" or "RequestId".

Any thoughts?

Reply:
I think "warning" is appropriate since this is a failure scenario. I'd expect "debug" to be normal successful stuff.

Reply:
For the code, is the formatting change intentional?

For the log message, add more context that just "Response: ~p" so that the log message makes sense on its own. Perhaps "Response from failed AWS API call: ~p". Think about how a user could use this in a log scan metric, in stream processing, etc.

Reply:
Thanks Stefan, I'll update the log message.

The formatting change is intentional. I responded to Thuan's comment on the same topic.